components/pagination/collections/paging_collection.js

```diff
@@ -0,0 +1,369 @@
+             * @param fieldName querystring parameter name for the
+             *     filterable field
+             */
+            filterDisplayName: function (fieldName) {
+                if (!_.isUndefined(this.filterFields[fieldName])) {
+                    return this.filterFields[fieldName].displayName;
+                } else {
+                    return '';
+                }
+            },
+
+            /**
+             * Sets the field to sort on and marks the collection as
+             * stale.
+             * @param fieldName name of the field to sort on
+             * @param toggleDirection if true, the sort direction is
+             *     toggled if the given field was already set
+             */
+            setSortField: function (fieldName, toggleDirection) {
+                var direction = toggleDirection ? 0 - this.state.order : this.state.order;
+                if (fieldName !== this.state.sortKey || toggleDirection) {
+                    this.setSorting(fieldName, direction);
+                    this.isStale = true;
+                }
+            },
+
+            /**
+             * Returns the direction of the current sort.
+             */
+            sortDirection: function () {
+                return (this.state.order === -1) ?
+                    PagingCollection.SortDirection.ASCENDING :
+                    PagingCollection.SortDirection.DESCENDING;
+            },
+
+            /**
+             * Sets the direction of the sort and marks the collection
+             * as stale.  Assumes (and requires) that the sort key has
+             * already been set.
+             * @param direction either ASCENDING or DESCENDING from
+             *     PagingCollection.SortDirection.
+             */
+            setSortDirection: function (direction) {
+                var currentOrder = this.state.order,
+                    newOrder = currentOrder;
+                if (direction === PagingCollection.SortDirection.ASCENDING) {
+                    newOrder = -1;
+                } else if (direction === PagingCollection.SortDirection.DESCENDING) {
+                    newOrder = 1;
+                }
+                if (newOrder !== currentOrder) {
+                    this.setSorting(this.state.sortKey, newOrder);
+                    this.isStale = true;
+                }
+            },
+
+            /**
+             * Flips the sort order.
+             */
+            flipSortDirection: function () {
+                this.setSorting(this.state.sortKey, 0 - this.state.order);
+                this.isStale = true;
+            },
+
+            /**
+             * Returns whether this collection has defined a given
+             * filterable field.
+             * @param fieldName querystring parameter name for the
+             *     filterable field
+             */
+            hasFilterField: function (fieldName) {
+                return !_.isUndefined(this.filterFields[fieldName]);
```

Comment:
Though, if you do want the existing behavior this would be cleaner with `_.has`.